AWSAuthSDK/Sources/AWSMobileClient/AWSMobileResults.swift

```diff
@@ -66,9 +66,76 @@ public enum AWSMobileClientError: Error {
+             .federationProviderExists(let message),
+             .groupExists(let message),
+             .guestAccessNotAllowed(let message),
+             .idTokenAndAcceessTokenNotIssued(let message),
+             .idTokenNotIssued(let message),
+             .identityIdUnavailable(let message),
+             .internalError(let message),
+             .invalidConfiguration(let message),
+             .invalidLambdaResponse(let message),
+             .invalidOAuthFlow(let message),
+             .invalidParameter(let message),
+             .invalidPassword(let message),
+             .invalidState(let message),
+             .invalidUserPoolConfiguration(let message),
+             .limitExceeded(let message),
+             .mfaMethodNotFound(let message),
+             .notAuthorized(let message),
+             .notSignedIn(let message),
+             .passwordResetRequired(let message),
+             .resourceNotFound(let message),
+             .scopeDoesNotExist(let message),
+             .securityFailed(let message),
+             .softwareTokenMFANotFound(let message),
+             .tooManyFailedAttempts(let message),
+             .tooManyRequests(let message),
+             .unableToSignIn(let message),
+             .unexpectedLambda(let message),
+             .unknown(let message),
+             .userCancelledSignIn(let message),
+             .userLambdaValidation(let message),
+             .userNotConfirmed(let message),
+             .userNotFound(let message),
+             .userPoolNotConfigured(let message),
+             .usernameExists(let message):
+            return message
+        }
+    }
+    
+    /// Utility class which maps a service error to `AWSMobileClientError`. If no mapping is available, returns the original error.
+    public static func getMobileError(for error: Error) -> Error {
+        if error._domain == AWSCognitoIdentityProviderErrorDomain {
+            if error._code == -3000 {
+                return AWSMobileClientError.deviceNotRemembered(message: "This device does not have an id, either it was never tracked or previously forgotten.")
+            }
+            let message = (error as NSError).userInfo["message"] as? String ?? "Error information not available."
+            let type = (error as NSError).userInfo["__type"] as? String ?? "Error type not available."
+            let mobileError = ErrorMappingHelper(errorCode: type, message: message, error: error as NSError)
+            return mobileError
+        }
+        return error
+    }
     
-    static func CognitoAuthErrorMappingHelper(error: AWSCognitoAuthClientErrorType) -> AWSMobileClientError {
+    public static func getMobileError(cognitoAuthError: AWSCognitoAuthClientErrorType) -> AWSMobileClientError {
+        return AWSMobileClientError.CognitoAuthErrorMappingHelper(error: cognitoAuthError)
+    }
+    
+    public static func CognitoAuthErrorMappingHelper(error: AWSCognitoAuthClientErrorType) -> AWSMobileClientError {
```

Comment:
Now that we're making this public, let's take the opportunity to rename it to something that doesn't sound like a class constructor. Maybe both of these could be:

```swift
public static func makeMobileClientError(from error: AWSCognitoAuthClientErrorType) -> AWSMobileClientError { ... }

public static func makeMobileClientError(fromErrorCode errorCode: String....

```

Reply:
Renamed, keeping it as static initializer.  `public static func makeMobileClientError(from error: Error) -> Error` returns an `Error` if not mapping is found so cannot be made into a init method.